### ISSUE TITLE
NavigationView SelectedItem indicator gone when toggling PaneDisplayMode

### ISSUE THREAD

**Issue by HO-COOH** [microsoft/microsoft-ui-xaml#9812]
### Describe the bug

As title suggest.

### Steps to reproduce the bug

1. Open winui3 gallery, go to NavigationView page
2. Go to the 6-th example, where there is an option to toggle pane position
3. Select a page, then set the Pane position to "Top mode", then toggle it back
Observe the indicator bar is now gone.

### Expected behavior

The indicator bar should be present.

### Screenshots

![navigationview](https://github.com/user-attachments/assets/13f431c9-d03b-48a3-962f-f99b7516a1d4)

### NuGet package version

WinUI 3 - Windows App SDK 1.5.5: 1.5.240627000

### Windows version

Windows 11 (22H2): Build 22621

**Comment by codendone** [comment]
Note: The issue is the blue selection pill. The background highlight for the selected item is still correctly there.

Interestingly, the blue selection pill seems to work fine for me on the "Track an Order" item, but not on the "Browse" or "Order History" items.

**Comment by HO-COOH** [comment]
> Note: The issue is the blue selection pill. The background highlight for the selected item is still correctly there.
> 
> Interestingly, the blue selection pill seems to work fine for me on the "Track an Order" item, but not on the "Browse" or "Order History" items.

Yep, this is so weird. Also, all the foot items (Account, Your Cart, Help) don't work
